### PR TITLE
Update from the Astropy package template

### LINKED ISSUE #169: Readme badges are outdated
Need to:

* Remove Appveyor badge
* Remove CircleCI badge
* Add Azure badge (unless you are replacing with Action, then add Action badge)

### PR DESCRIPTION
**Pull request by lpsinger**
Fixes #169.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,17 +1,21 @@
 [tox]
 envlist =
-    py{36,37,38,39}-test{,-alldeps,-devdeps}{,-cov}
-    py{36,37,38,39}-test-numpy{116,117,118}
-    py{36,37,38,39}-test-astropy{30,40,lts}
+    py{36,37,38}-test{,-alldeps,-devdeps}{,-cov}
+    py{36,37,38}-test-numpy{116,117,118}
+    py{36,37,38}-test-astropy{30,40,lts}
```
Comment (astrofrog) [inline tox.ini:5]: I think we should add back 39 here

File: astropy_healpix/conftest.py [hunk 2]
```diff
@@ -37,3 +49,16 @@ def pytest_configure(config):
...
+# as follow (although default should work for most cases).
+# To ignore some packages that produce deprecation warnings on import
+# (in addition to 'compiler', 'scipy', 'pygments', 'ipykernel', and
+# 'setuptools'), add:
+#     modules_to_ignore_on_import=['module_1', 'module_2']
+# To ignore some specific deprecation warning messages for Python version
+# MAJOR.MINOR or later, add:
+#     warnings_to_ignore_by_pyver={(MAJOR, MINOR): ['Message to ignore']}
+# from astropy.tests.helper import enable_deprecations_as_exceptions  # noqa
+# enable_deprecations_as_exceptions()
```
Comment (astrofrog) [inline astropy_healpix/conftest.py:64]: Remove this block - it's going to be removed in the template now that this is deprecated in astropy core

**Review by astrofrog — COMMENTED** [review]
Just a couple of comments below

**Review by pllim — COMMENTED** [review]
I would defer to @astrofrog because I don't really maintain this package. But if you need second opinions, feel free to ping me again. Thanks!

File: tox.ini (hunk 1 above)
Comment (lpsinger) [reply tox.ini]: Done.

File: astropy_healpix/conftest.py (hunk 2 above)
Comment (lpsinger) [reply astropy_healpix/conftest.py]: Done.

**Review by astrofrog — CHANGES_REQUESTED** [review]
This looks good generally but also removes the auto-publishing of releases - could you add this back in the Azure config or alternatively add equivalent GitHub actions config?

**Comment by lpsinger** [comment]
> This looks good generally but also removes the auto-publishing of releases - could you add this back in the Azure config or alternatively add equivalent GitHub actions config?

Done.

Would you please make me (https://pypi.org/user/lsinger/) a co-owner of this project on PyPI so that I can create an API token for GitHub Actions secrets?

**Comment by lpsinger** [comment]
> > This looks good generally but also removes the auto-publishing of releases - could you add this back in the Azure config or alternatively add equivalent GitHub actions config?
> 
> Done.
> 
> Would you please make me (https://pypi.org/user/lsinger/) a co-owner of this project on PyPI so that I can create an API token for GitHub Actions secrets?

@astrofrog, @cdeil?

**Comment by lpsinger** [comment]
@astrofrog?

**Review by astrofrog — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -64,10 +64,6 @@ pip-wheel-metadata/
 # Mac OSX
 .DS_Store
 
-astropy_healpix/_compiler.c
```
Comment (pllim) [inline .gitignore:67]: Why are these un-ignored? Are they no longer generated?

Comment (lpsinger) [reply .gitignore:67]: That's right.

**Pull request merged**